### PR TITLE
Modules API basic fixes + improvements

### PR DESCRIPTION
**Pull request by RinesThaix**

### CODE REVIEW

File: module-api/src/main/java/ru/feathercore/moduleapi/ModuleLoader.java [hunk 1]
```diff
@@ -62,4 +62,15 @@
      * @apiNote modifications to returned collection should not have side-effect on this module loader
      */
     Collection<? extends M> getModules();
+
+    /**
+     * Checks whether module of given class was loaded by this module loader.
```
Comment (JarvisCraft) [inline module-api/src/main/java/ru/feathercore/moduleapi/ModuleLoader.java]: Checks whether *there is a* module of given *type* *currently* loaded by this module loader.`

File: module-api/src/main/java/ru/feathercore/moduleapi/ModuleLoader.java [hunk 2]
```diff
@@ -62,4 +62,15 @@
...
+     *
+     * @param module class of module to be checked.
+     * @return true/false.
+     */
+    default boolean isModuleLoaded(Class<? extends M> module) {
```
Comment (JarvisCraft) [inline module-api/src/main/java/ru/feathercore/moduleapi/ModuleLoader.java]: -> `isLoaded(Class<? extends M> module)`

File: module-api/src/main/java/ru/feathercore/moduleapi/ModuleLoader.java [hunk 3]
```diff
@@ -62,4 +62,15 @@
...
+        return getModules().stream().anyMatch(mod -> module.isAssignableFrom(mod.getClass()));
```
Comment (JarvisCraft) [inline module-api/src/main/java/ru/feathercore/moduleapi/ModuleLoader.java]: Provide default implemention in `SimpleModuleLoader`

**Review by JarvisCraft — CHANGES_REQUESTED** [review]

File: module-api/src/main/java/ru/feathercore/moduleapi/ModuleLoader.java (hunk 3 above)
Comment (RinesThaix) [reply module-api/src/main/java/ru/feathercore/moduleapi/ModuleLoader.java]: Why?

Comment (JarvisCraft) [reply module-api/src/main/java/ru/feathercore/moduleapi/ModuleLoader.java]: getModules() is contracted to return unmodifiable view of loaded modules and so some implementations may provide a more efficient implementation. A default one may be kept, but an overriden equivalent should definitely be added to default implementation

File: module-api/src/main/java/ru/feathercore/moduleapi/ModuleLoader.java [hunk 4]
```diff
@@ -62,4 +62,15 @@
      * @apiNote modifications to returned collection should not have side-effect on this module loader
      */
     Collection<? extends M> getModules();
+
+    /**
+     * Checks whether module of given class was loaded by this module loader.
+     *
+     * @param module class of module to be checked.
+     * @return true/false.
```
Comment (JarvisCraft) [inline module-api/src/main/java/ru/feathercore/moduleapi/ModuleLoader.java]: Provide more exact description for `@return` also using `{@link true}` and `{@link false}`

**Review by JarvisCraft — CHANGES_REQUESTED** [review]

**Review by JarvisCraft — APPROVED** [review]

**Pull request merged**